### ISSUE TITLE
Support Rails 7.1

### ISSUE THREAD

**Issue by cprodhomme** [lynndylanhurley/devise_token_auth#1608]
* **Version**: 1.2.2

Since the 13 october, Rails release the version [7.1.0.beta1](https://rubygems.org/gems/rails/versions/7.1.0.beta1).

I thinks the official release of Rails 7.1 will be soon with amazing features !
Could you support this version ?

i saw some works on this PR : https://github.com/lynndylanhurley/devise_token_auth/pull/1606

Thanks ! 🙏

**Comment by scott-knight** [comment]
Would be nice to have an ETA on this.

**Comment by BenMukebo** [comment]
We're  still waiting for the update cause currently I can't using devise_token_auth in th lastest vesion of ruby on rails!!

**Comment by dhaigh** [comment]
sooner the better, 7.1.0 has been out since October 5 🙏

**Comment by smenor** [comment]
just a 「 me too 」

**Comment by up_the_irons** [comment]
I also want to use devise-token-auth but my new app is Rails 7.1.1 😭

**Comment by up_the_irons** [comment]
I found this fork: https://github.com/single-stop-tony/devise_token_auth

It makes a slight change to a part of the serialization and updated the gem spec to allow installation with Rails 7.1.  No idea if it *actually* works now, or is a work in progress, but I'm going to give it a try.

**Comment by rohanmendon** [comment]
same here, want to use it on rails 7.1.2

**Comment by lsfernandes92** [comment]
I had the same error here and it seems that [this stack overflow answer](https://stackoverflow.com/a/71088408) addressed my question regarding this issue

**Comment by orehmane** [comment]
That's a fine workaround, but it would nice to have a more proper version in rubygems.

**Comment by angusm73** [comment]
Adding the gem from github instead of rubygems does indeed work!
```
gem "devise_token_auth", ">= 1.2.0", git: "https://github.com/lynndylanhurley/devise_token_auth"
```

**Comment by scott-knight** [comment]
What's the hold up here? I can't get graphql_devise upgraded without this update.

**Comment by groue** [comment]
Hello. I'm also waiting for Rails 7.1 for deferred foreign keys support.

**Comment by hatsu38** [comment]
@MaicolBen Thank you for your continuous maintenance. I really appreciate the convenience it provides!

I believe this ISSUE will be resolved once the version containing this PR #1606  is released. 
Is there anything missing for the release?
If so, I am willing to help. Many people are looking forward to the release! Please make it happen!

**Comment by arp** [comment]
Same here. There are a few fixes that are already in master, so I had to port them manually in our app. Is there anything I could help with to release the new gem version?

**Comment by smenor** [comment]
Just adding a 「 me-too 」

**Comment by StalemateInc** [comment]
Joining the flock of requesters - a stable version of Rails 7.1 was published four months ago and for some people explicit requirement of `rails < 7.1` is a roadblock for upgrading. Can we at least extract the fixes for Rails 7.1 and have a patch level release?

**Comment by ManCodeB** [comment]
+1 from me to support Rails 7.1 🙂 

This gem blocks me from upgrading - thank you in advance!

**Comment by saurabhbhatia** [comment]
+1 on this, would really appreciate any movement on this

**Comment by underscorewilian** [comment]
Joining the waiting crew, with a side of anxiety. 🫤😬

**Comment by jonathan-at-leavelogic** [comment]
I would greatly appreciate a status update on this gem, too, as it's affecting Rails 7.1 upgrade efforts.

**Comment by jpayne** [comment]
I would greatly appreciate an update as well, so that Rails 7.1 upgrades can be completed.

**Comment by kenoga** [comment]
👍🏼

**Comment by VSidhArt** [comment]
Bump version, please
We're all waiting for this update. It's blocked our rails upgrade.

**Comment by scott-knight** [comment]
I finally gave up waiting and created a [fork](https://github.com/scott-knight/devise_token_auth). I'm not sure what the holdup is, but anyone is welcome to use the fork until they do their release.

**Comment by mgmodell** [comment]
It's been 5 months since any commits on this project and this blocker issue is open for about 6 months now. At what point does this get classified as _abandoned_? I, too, have been using my own fork, but a billion of those is not ideal. What do we do to reinvigorate this project?

**Comment by lynndylanhurley** [comment]
@mgmodell @scott-knight would either of you guys be interested in commit access to help push new releases?

**Comment by mgmodell** [comment]
I would be happy to support - as long as someone can be available to answer my questions as I work through it.

**Comment by mgmodell** [comment]
@lynndylanhurley, I'm hoping that we can move forward with this and I'll gladly take on the tasks, but the recent [xz issue](https://arstechnica.com/security/2024/04/what-we-know-about-the-xz-utils-backdoor-that-almost-infected-the-world/) leaves me wanting oversight to make sure we can maintain the confidence of the community. Can we please move forward with this? Perhaps @scott-knight would like to share whatever load there might be? I've certainly got a lot to learn but there are clearly lots of people relying upon this software and I'll happily do my part (as my copious free time permits).

**Comment by scott-knight** [comment]
@mgmodell - I'm wondering if @lynndylanhurley as fallen off the earth. Seems unusual for something like this to go on as long as it has -- releases in the past have been consistent.  Anyone know the status of @lynndylanhurley?

**Comment by mgmodell** [comment]
@scott-knight I'm sure he'll pop back on when he gets a moment. This was posted just last week. Great that @lynndylanhurley is busy - that's got to be a good thing. Last week's post indicates he's still monitoring and wants to be sure this is in good hands and, especially since this is security-related and I've not gone over every bit of code in this repo or ever maintained a gem before, I'm up for being patient if it means doing it right. We've waited 5 months - a little more is acceptable to me, given the subject matter.

**Comment by mgmodell** [comment]
@lynndylanhurley - can we make this happen?

**Comment by MaicolBen** [comment]
I bumped 1.2.3, is there anything else missing? Because I don't understand why people were forking when you can use this repo, just checked a fork so not sure if I missed something

**Comment by MaicolBen** [comment]
> At what point does this get classified as abandoned?

We could but a huge amount of people are relying on it, we need maintainers (we have a bounties and can give a little $ to people willing to help), and my bandwidth is only for reviewing PRs and bumping versions. Feel free to file an issue with this and suggest replacements so we don't this divert this rails issue.

**Comment by mgmodell** [comment]
I do have a fork that I rely upon and I made the required changes on my end, so I was OK, but I believe that many rely upon the published gem being updated and it was not. This prevented many of those who rely upon this gem from upgrading their version of rails to 7.1. I would be happy to help out with these sorts of tasks if you will provide me with whatever access would be required. Ideally, I would also have someone of whom I could ask questions when they are beyond me.
I see that many are relying upon it and I'd like to help make it possible for them to continue to do so. I do not want it to be abandoned, but there's been almost no response from anyone who could act since September. I will gladly help if you'll make it possible for me to do so.

**Comment by hatsu38** [comment]
The branch for Rails 7.2 has been created and it is expected to be released soon. 

I am planning to work on the compatibility of devise_token_auth with Rails 7.2 next. 

It seems that many people were waiting for this gem to be released for Rails 7.1, so it would be great if we could release it without making everyone wait too long for Rails 7.2!

**Comment by MaicolBen** [comment]
> I will gladly help if you'll make it possible for me to do so.

Feel free to create PRs for fixing the issues we have! After a while, we can give you more access. I will keep an eye more often in version bumping, I didn't realize some people cannot point to a github repo. 

> I am planning to work

@hatsu38 Thanks, let me know if you cannot file a PR so I can do it instead. 

I am closing this issue, please continue commenting here if you want to add thoughts or feel this issue should still be open
